synapse/lib/syntax.py

```diff
@@ -1495,6 +1502,24 @@ def nextstr(self, text):
         subt = self.text[self.offs:self.offs + size]
         return subt == text
 
+    def nextstrs(self, *texts):
+        if not self.more():
+            return False
+        sd = self.getSortedDict(*texts)
+        for size, valus in sd.items():
+            subt = self.text[self.offs:self.offs + size]
+            if subt in valus:
+                return True
+        return False
+
+    @s_cache.memoize()
+    def getSortedDict(self, *texts):
+        d = collections.defaultdict(list)
+        for text in texts:
+            d[len(text)].append(text)
+        ret = {k: d.get(k) for k in sorted(d.keys())}
```

Comment:
`{k: d[k] ...` is a little clearer to me (since it is double impossible that d[k] isn't set).